app/controllers/api/messaging.php

```diff
@@ -3251,19 +3251,64 @@
             throw new Exception(Exception::MESSAGE_NOT_FOUND);
         }
 
+        if ($status !== MessageStatus::DRAFT && \count($topics) === 0 && \count($users) === 0 && \count($targets) === 0) {
+            throw new Exception(Exception::MESSAGE_MISSING_TARGET);
+        }
+
+        $currentScheduledAt = $message->getAttribute('scheduledAt');
+
         switch ($message->getAttribute('status')) {
             case MessageStatus::PROCESSING:
                 throw new Exception(Exception::MESSAGE_ALREADY_PROCESSING);
             case MessageStatus::SENT:
                 throw new Exception(Exception::MESSAGE_ALREADY_SENT);
             case MessageStatus::FAILED:
                 throw new Exception(Exception::MESSAGE_ALREADY_FAILED);
+            case MessageStatus::SCHEDULED:
+                if (\is_null($scheduledAt) && \is_null($currentScheduledAt)) {
+                    throw new Exception(Exception::MESSAGE_MISSING_SCHEDULE);
+                }
+                break;
         }
 
-        if (!\is_null($message->getAttribute('scheduledAt')) && $message->getAttribute('scheduledAt') < new \DateTime()) {
+        if (!\is_null($currentScheduledAt) && $currentScheduledAt < new \DateTime()) {
             throw new Exception(Exception::MESSAGE_ALREADY_SCHEDULED);
         }
 
+        if (\is_null($currentScheduledAt && !\is_null($scheduledAt))) {
+            $schedule = $dbForConsole->createDocument('schedules', new Document([
+                'region' => App::getEnv('_APP_REGION', 'default'),
+                'resourceType' => 'message',
+                'resourceId' => $message->getId(),
+                'resourceInternalId' => $message->getInternalId(),
+                'resourceUpdatedAt' => DateTime::now(),
+                'projectId' => $project->getId(),
+                'schedule' => $scheduledAt,
+                'active' => $status === MessageStatus::SCHEDULED,
+            ]));
+
+            $message->setAttribute('scheduleId', $schedule->getId());
+        }
+
+        if (!\is_null($currentScheduledAt)) {
+            $schedule = $dbForConsole->getDocument('schedules', $message->getAttribute('scheduleId'));
+
+            if ($schedule->isEmpty()) {
+                throw new Exception(Exception::SCHEDULE_NOT_FOUND);
+            }
+
+            if (!\is_null($scheduledAt)) {
+                $schedule
+                    ->setAttribute('resourceUpdatedAt', DateTime::now())
+                    ->setAttribute('schedule', $scheduledAt)
+                    ->setAttribute('active', $status === MessageStatus::SCHEDULED);
+            } else {
+                $schedule->setAttribute('active', $status === MessageStatus::SCHEDULED);
```

Comment:
Should this be the same as the above?

Reply:
No, if you didn't changed the scheduled at time, it should just be activated/deactivated